tests/models/seamless_m4t/test_feature_extraction_seamless_m4t.py

```diff
@@ -62,7 +62,7 @@ def __init__(
         feature_size=10,
         padding_value=0.0,
         sampling_rate=4_000,
-        return_attention_mask=True,
+        return_attention_mask=False,
```

Comment:
It's tested in [`SequenceFeatureExtractionTestMixin.test_attention_mask`](https://github.com/huggingface/transformers/blob/4a66c0d95219bbeb91bebd010d75a29a5e5f3f43/tests/test_sequence_feature_extraction_common.py#L389-L424), but I can still add some more tests if you want!

Reply:
These tests only test the `.pad` call to the feature extractor. Not the `attention_mask` logic when calling the feature extractor directly - which this PR addresses

Reply:
You're right, I've added a test

Reply:
Let me know if that works for you @amyeroberts !

Reply:
Let's take a step back so I can understand the changes: 
* why is the default being changed here? 
* what is the default behaviour of the feature extractor (outside of tests)? Does it return the attention masks by default?

Reply:
I've realized when copying the M4T feature extractor for another model there were errors when `return_attention_mask=False`, I've thus corrected the feature extractor code.

Now, with regards to tests, I realized it wasn't flagged by test so:

1. I've changed the default to `False`, because no current tests were actually using the attention mask.
2. The new test `test_call_attention_mask` checks that one can use the attention mask without errors.

I've done that despite the default behavior outside of tests being `return_attention_mask=True`.

Let me know if that responds to your questions. I can still change the default back to `True` and test without attention mask similarly to `test_call_attention_mask` if that's needed.

Reply:
OK, so in a perfect world we'd have full coverage of the behaviour of our objects, in this case with all different argument permutations and their expected output. As this isn't practical, it's important that the most common use cases are accounted for. All class methods are effectively black boxes: I do not know if the different input arguments interact or not. Therefore, it's easier and better when testing to use the defaults and interrogate the outputs based on changing a subset of the arguments.

Reply:
To be clear, a test should be added to check everything works when `return_attention_mask=False`, but it should be left as `True` here to make sure the default behaviour is compatible with what's happening in other tests

Reply:
This makes total sense, thanks for the explanations here!  
I've modified the tests to go back to the initial situation but I've added two other tests to ensure better coverage.

I'll make sure to keep your explanation in mind for future PRs as well !